### PR TITLE
Autotests: #3168 - Open and Save files (Mol-files)

### PR DESCRIPTION
**Pull request by OlgaMazurina**
## How the feature works? / How did you fix the issue?
Automated 15 tests for Open and Save files. Some of them have bugs, so was added `fixme`

## Check list
- [ ] unit-tests written
- [x] e2e-tests written
- [ ] documentation updated
- [x] PR name follows the pattern `#1234 – issue name`
- [x] branch name doesn't contain '#'
- [x] PR is linked with the issue
- [x] base branch (master or release/xx) is correct
- [x] task status changed to "Code review"
- [x] reviewers are notified about the pull request

### CODE REVIEW

**Review by chgayane — APPROVED** [review]

**Comment by OlgaMazurina** [comment]
All tests passed
Снимок экрана 2023-08-28 в 13 11 15

**Review by y-holik — APPROVED** [review]

File: ketcher-autotests/tests/test-data/rgroup-V3000-expected
Comment (Nitvex) [inline ketcher-autotests/tests/test-data/rgroup-V3000-expected:1]: Can we move this file to Molfiles-V3000 folder?

File: ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts [hunk 1]
```diff
@@ -538,3 +538,508 @@ test('Open V3000 file with R-Groups with Fragments', async ({ page }) => {
...
+   * Test case: EPMLSOPKET-1852
+   * Description: Structure with the correct Bond properties is correctly represented
+   */
+  await page.goto('');
+
+  await openFileAndAddToCanvas('Molfiles-V2000/all_bond_properties.mol', page);
+  await takeEditorScreenshot(page);
+});
+
+test.fixme(
```
Comment (Nitvex) [inline ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts:555]: If test doesn't for some reason, could you, please, add a comment here?

File: ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts [hunk 2]
```diff
@@ -538,3 +538,508 @@ test('Open V3000 file with R-Groups with Fragments', async ({ page }) => {
...
+test('Open and Save file - Open/Save files for ferrocen-like structures 1/2 - open', async ({
+  page,
+}) => {
+  /**
+   * Test case: EPMLSOPKET-1893(1)
+   * Description: Structures are rendered correctly
+   */
+  await page.goto('');
+
+  await openFileAndAddToCanvas('Molfiles-V2000/ferrocene_radical01.mol', page);
```
Comment (Nitvex) [inline ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts:822]: Could you, please, rename the files to follow the convention?
`Molfiles-V2000/ferrocene-radical-02.mol`

May be it is wort to add a loop?
```
for (let i=1; i<9; i++) {
 await openFileAndAddToCanvas(`Molfiles-V2000/ferrocene-radical0${i}.mol`, page);
}
```

And actually, I don't understand the test – you add multiple structures above one another and then do the screenshot of the result. May be it is better to add structure one by one, make a screenshot and then clear the canvas ?

**Review by Nitvex — CHANGES_REQUESTED** [review]

File: ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts (hunk 2 above)
Comment (OlgaMazurina) [reply ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts:822]: Now it looks good, tnx!

File: ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts [hunk 3]
```diff
@@ -538,3 +538,508 @@ test('Open V3000 file with R-Groups with Fragments', async ({ page }) => {
...
 });
+
+test('Open and Save files - Open/Save structure with bond properties', async ({
+  page,
+}) => {
+  /**
+   * Test case: EPMLSOPKET-1852
+   * Description: Structure with the correct Bond properties is correctly represented
+   */
+  await page.goto('');
```
[the remark this answers is not in the document]
Comment (OlgaMazurina) [reply ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts:549]: Good idea, now it is better

File: ketcher-autotests/tests/test-data/rgroup-V3000-expected
Comment (OlgaMazurina) [reply ketcher-autotests/tests/test-data/rgroup-V3000-expected:1]: Yes, it's a vestige

File: ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts [hunk 4]
```diff
@@ -538,3 +538,508 @@ test('Open V3000 file with R-Groups with Fragments', async ({ page }) => {
...
+  async ({ page }) => {
+    /**
+     * Test case: EPMLSOPKET-1875(2)
+     * Description: Structure is correctly generated from Molstring
+     * and vise versa molstring is correctly generated from structure.
+     * */
+
+    await page.goto('');
+
+    await openFileAndAddToCanvas('Molfiles-V2000/sgroup_different.mol', page);
```
[the remark this answers is not in the document]
Comment (OlgaMazurina) [reply ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts]: Did it

File: ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts [hunk 5]
```diff
@@ -547,3 +549,268 @@ test('Open V3000 file with R-Groups with Fragments', async ({ page }) => {
   await openFileAndAddToCanvas('RGroup-With-Fragments.mol', page);
   await takeEditorScreenshot(page);
 });
+
+test.describe('', () => {
```
Comment (Nitvex) [inline ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts:553]: I propose to add some meaningful string for describe

File: ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts [hunk 6]
```diff
@@ -547,3 +549,268 @@ test('Open V3000 file with R-Groups with Fragments', async ({ page }) => {
...
+    const files = [
+      {
+        testName:
+          'Open and Save files - Open/Save structure with bond properties',
+        path: 'Molfiles-V2000/all-bond-properties.mol',
+      },
+      {
+        testName:
+          'Open and Save file - Open/Save Alias and Pseudoatoms 1/2 - open',
+        path: 'Molfiles-V2000/mol-1856-to-open.mol',
```
Comment (Nitvex) [inline ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts:573]: Could you, please, add more meaningful names for files, describing their contents? May be some files are not even needed. Sometimes in test-cases they have different names, but their content is the same

File: ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts [hunk 7]
```diff
@@ -547,3 +549,268 @@ test('Open V3000 file with R-Groups with Fragments', async ({ page }) => {
...
+          'tests/test-data/Molfiles-V2000/marvin-bond-properties-V3000-expected.mol',
+        fileFormat: 'v2000',
+        metaDataIndexes: METADATA_STRING_INDEX,
+      });
+
+    expect(molFile).toEqual(molFileExpected);
+  });
+
+  for (let i = 1; i < 9; i++) {
+    test(`'Open and Save file - Open/Save files for ferrocen-like structures 1/2 - open ferrocene_radical0${i}.mol`, async ({
```
Comment (Nitvex) [inline ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts:749]: guess you can remove `'` in test name :)

File: ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts (hunk 5 above)
Comment (OlgaMazurina) [reply ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts]: Done

File: ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts (hunk 7 above)
Comment (OlgaMazurina) [reply ketcher-autotests/tests/File-Management/Open-And-Save-Files/MOL-Files/mol-files.spec.ts]: Thank you, fixed

**Review by Nitvex — APPROVED** [review]

**Pull request merged**